src/core/elf.c

```diff
@@ -40,40 +326,212 @@ elf_header_t* elf_load(uint64_t* data) {
     return elf;
 }
 
+void elf_unload(elf_header_t* elf) {
+
+    DEBUG("%s", "Unloading an elf file");
+
+    //Note: This does not currently work
```

Comment:
Then we might want to remove it for now

Reply:
Yeah, I'll keep this PR closed as a reference and then do this in separate waves of PRs. Doesn't make sense as it is and as you said, it grew too much.